ttnn/cpp/ttnn/multi_device.hpp

```diff
@@ -44,9 +43,12 @@ std::vector<ttnn::Tensor> get_device_tensors(const ttnn::Tensor& tensor) {
         return tensors;
     } else if (std::holds_alternative<tt::tt_metal::MultiDeviceStorage>(tensor.get_storage())) {
         auto& device_storage = std::get<tt::tt_metal::MultiDeviceStorage>(tensor.get_storage());
-        for (int i = 0; i < device_storage.buffers.size(); ++i)
+        std::vector<ttnn::Tensor> tensors = std::vector<ttnn::Tensor>(device_storage.buffers.size(), Tensor());
+        for (auto& device_buf_pair : device_storage.buffers)
         {
-            tensors.push_back(Tensor{DeviceStorage{device_storage.buffers[i]}, device_storage.shapes[i], tensor.get_dtype(), tensor.get_layout()});
+            auto buffer = device_buf_pair.second;
+            auto device = device_buf_pair.first;
+            tensors[device->id()] = Tensor{DeviceStorage{buffer}, device_storage.shapes.at(device), tensor.get_dtype(), tensor.get_layout()};
```

Comment:
Do we need the `Tensor` to be created through the constructor path now?

Reply:
Constructor usage for the tensor is unchanged here. We need to use it here, since we initializing a non-empty tensor in the main thread.